pandas/tests/series/methods/test_fillna.py

```diff
@@ -946,3 +947,27 @@ def test_datetime64tz_fillna_round_issue(self):
         )
 
         tm.assert_series_equal(filled, expected)
+
+    def test_fillna_parr(self):
+        # GH-24537
+        dti = date_range(
+            Timestamp.max - Timedelta(nanoseconds=10), periods=5, freq="ns"
+        )
+        pi = dti.to_period("ns")
```

Comment:
You can do ser = Series(dti.to_period("ns")) here and then do

ser[2] = NaT without accessing _data in the line below, makes it a bit simpler.

And then do ser.fillna(method="pad")

Reply:
Ah ok thanks will try to change it 👍